service/dap/server.go

```diff
@@ -2334,8 +2334,13 @@ func (s *Server) onExceptionInfoRequest(request *dap.ExceptionInfoRequest) {
 	if bpState != nil && bpState.Breakpoint != nil && (bpState.Breakpoint.Name == proc.FatalThrow || bpState.Breakpoint.Name == proc.UnrecoveredPanic) {
 		switch bpState.Breakpoint.Name {
 		case proc.FatalThrow:
-			// TODO(suzmue): add the fatal throw reason to body.Description.
 			body.ExceptionId = "fatal error"
+			// Attempt to get the value of the throw reason.
+			// This is not currently working for Go 1.16 or 1.17: https://github.com/golang/go/issues/46425.
```

Comment:
That's a good idea! I will add that as the reason if its on Go 1.16 or Go 1.17.

I have been considering these additional exception info descriptions best effort, if they don't work, we just don't include them. I think it would be good now to include these extra info messages so that it can help us know if it is no longer working easier.